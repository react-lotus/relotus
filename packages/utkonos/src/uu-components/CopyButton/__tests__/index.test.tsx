import { render, screen, fireEvent } from '@testing-library/react';

import { CopyButton } from '..';

const valueToRaiseMockException = 'fake input causing exception in copy to clipboard';
jest.mock('copy-to-clipboard', () =>
  jest.fn().mockImplementation((input: string) => {
    if (input === valueToRaiseMockException) {
      throw new Error(input);
    }
    return true;
  }),
);

const onClick = jest.fn();

describe('Copy Button component', () => {
  it('render correctly', () => {
    expect(() => render(<CopyButton role="button" value="text" />)).not.toThrow();
  });

  it('should render square button by default', () => {
    render(<CopyButton role="button" value="text" />);
    const button = screen.queryByRole('button');
    expect(button).toHaveClass('uu-button_square');
    expect(button).toMatchSnapshot();
  });

  it('should render as tag <button /> by default', () => {
    render(<CopyButton role="button" value="text" />);
    const button = screen.queryByRole('button');
    expect(button?.tagName).toBe('BUTTON');
  });

  it('should render as light button by default', () => {
    render(<CopyButton role="button" value="text" />);
    const button = screen.queryByRole('button');
    expect(button).toHaveClass('uu-button_view_light');
  });

  it('should call copy to clipboard with text', () => {
    render(<CopyButton role="button" onClick={onClick} value="text" />);
    const copyButton = screen.queryByRole('button');
    if (copyButton) {
      fireEvent.click(copyButton);
    }
    expect(onClick).toHaveBeenCalled();
  });
});
