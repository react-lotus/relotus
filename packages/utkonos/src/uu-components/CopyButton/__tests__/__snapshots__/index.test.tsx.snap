// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copy Button component should render square button by default 1`] = `
<button
  class="uu-button uu-button_view_light uu-button_square"
  role="button"
  type="button"
>
  <svg>
    copy-icon.svg
  </svg>
</button>
`;
